src/cpp/session/modules/SessionCodeTools.R

```diff
@@ -2587,3 +2587,33 @@
    getNamespace(envir)
    
 })
+
+.rs.addJsonRpcHandler("project_guess_cpp_style", function(inDirectory = .rs.getProjectDirectory()) 
+{
+    # if there is no DESCRIPTION file, can't guess
```

Comment:
cc @jennybc @sbearrows you might have views on how to improve the guessing logic.

Reply:
The Rcpp vs cpp11 guessing logic looks sound to me.

With the caveat that I have basically zero feel for the conventions and style of the R code here, I would find this function easier to understand if it was refactored to have more early exits. For example, you could bring the "we have no basis for a guess" case up to the very top, i.e. this branch:

```
# otherwise just use the default from user preferences 
.rs.scalar("")
```

This would also make it easy to pull the `DESCRIPTION <- ...` assignment out of the condition, which I would find more readable.

Reply:
Thanks :-)